pyadjoint/checkpointing.py

```diff
@@ -180,16 +187,15 @@ def recompute(self, functional=None):
         Args:
             functional (BlockVariable): The functional to be evaluated.
         """
+        if self.mode == Mode.RECORD:
+            # Finalise the taping process.
+            self.end_taping()
         self.mode = Mode.RECOMPUTE
-
-        with self.tape.progress_bar("Evaluating Functional",
-                                    max=self.timesteps) as bar:
-            # Restore the initial condition to advance the forward model
-            # from the step 0.
+        with self.tape.progress_bar("Evaluating Functional", max=self.total_timesteps) as bar:
+            # Restore the initial condition to advance the forward model from the step 0.
             current_step = self.tape.timesteps[self.forward_schedule[0].n0]
             current_step.restore_from_checkpoint()
             for cp_action in self.forward_schedule:
-                self._current_action = cp_action
```

Comment:
Why I made this?

Reply:
I think I should not remove this line. `self._current_action` is not being used in recompute forward, but `self._current_action` should always updated.